File: backend/Models/Lead.cs
using System.ComponentModel.DataAnnotations;
using backend.Models.QuoteModels;

namespace backend.Models;

public class Lead
{
    [Key]
    public Guid Id { get; set; }
    public Guid? CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Source { get; set; }
    public Guid? QuoteId { get; set; }
    public Quote? Quote { get; set; }
    public Guid WorkspaceId { get; set; }
    public Workspace? Workspace { get; set; }
    [Required]
    public string Description { get; set; } = string.Empty;
    public DateTime? StartDateTime { get; set; }
    public DateTime? EndDateTime { get; set; }
    public LeadStatus Status { get; set; } = LeadStatus.Pending;
    public LeadPriority Priority { get; set; } = LeadPriority.Normal;
    public List<LineItem> LineItems { get; set; } = [];
    public bool IsArchived { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}


public enum LeadStatus
{
    Pending,
    Reviewed,
    Approved,
    Declined,
    Converted
}

public enum LeadPriority
{
    Low,
    Normal,
    High,
    Urgent
}

File: frontend/src/types/Lead.ts
import { TCustomer } from './Customer';
import { TWorkspace } from './Workspace';
import { TAddLineItem, TLineItem } from './LineItem';
import { TQuote } from './Quote';
import {
  LeadPriority,
  LeadStatus,
} from '../constants/Enumeration/LeadEnum/LeadEnum';

export type TLead = {
  id: string;
  customerId?: string;
  customer?: TCustomer;
  quoteId?: string;
  quote?: TQuote;
  workspaceId: string;
  workspace?: TWorkspace;
  firstName?: string;
  lastName?: string;
  email?: string;
  phoneNumber?: string;
  source?: string;
  description: string;
  startDateTime?: string;
  endDateTime?: string;
  status: LeadStatus;
  priority: LeadPriority;
  lineItems: TLineItem[];
  notes?: string;
  createdAt: string;
  updatedAt: string;
};

export type TAddLead = {
  customerId?: string;
  workspaceId: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  phoneNumber?: string;
  source?: string;
  description: string;
  startDateTime?: string;
  endDateTime?: string;
  priority: LeadPriority;
  lineItems: TAddLineItem[];
  notes?: string;
};

export type TUpdateLead = {
  id: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  phoneNumber?: string;
  source?: string;
  description?: string;
  startDateTime?: string;
  endDateTime?: string;
  status?: LeadStatus;
  priority?: LeadPriority;
  lineItems?: TAddLineItem[];
  notes?: string;
};
